### PR TITLE
Add RELEASE.md

### PR DESCRIPTION
**Pull request by st--**
Adapted from TensorFlow's running release notes.

Also includes release notes since the last release (2.1.4).

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1 +1,64 @@
...
+* <NOTES SHOULD BE GROUPED PER AREA>
+
+## Thanks to our Contributors
+
+This release contains contributions from:
+
+<INSERT>, <NAME>, <HERE>, <USING>, <GITHUB>, <HANDLE>
+
+
+# Release 2.1.5
```
Comment (vdutor) [inline RELEASE.md:36]: can you make it more clear this is the current release people have to add notes to?

File: RELEASE.md [hunk 2]
```diff
@@ -1 +1,64 @@
-The release notes have been moved to the ['Releases' section](https://github.com/GPflow/GPflow/releases) of the GPflow GitHub Repo
+Release notes for past releases are available in the ['Releases' section](https://github.com/GPflow/GPflow/releases) of the GPflow GitHub Repo.
+
+# Release x.y.z
```
Comment (vdutor) [inline RELEASE.md:3]: can we not have the template at the bottom of the page?

Comment (st--) [reply RELEASE.md:3]: We could, but that makes it more hassle when creating a new release to copy things from the bottom back to the top - to me it makes more sense to follow TensorFlow in this and leave as is.

File: RELEASE.md (hunk 1 above)
Comment (st--) [reply RELEASE.md]: done

**Review by vdutor — APPROVED** [review]
LGTM. Can you add an extra step to our release procedure that the person in charge of making the release also copies the template and starts the release notes for the next release?

**Comment by st--** [comment]
Great idea - I added it as a HOWTO_RELEASE.md document to the repo so everyone can access it & propose edits.

**Pull request merged**